backend/tests/test_rate_limiting.py

```
@@ -0,0 +1,295 @@
+            response = client.post('/api/auth/login', json=login_data)
+            
+            if response.status_code == 429:  # Rate limited
+                # Rate limiting is working
+                assert attempt >= 3, "Rate limiting should kick in after a few attempts"
+                break
+            elif response.status_code == 401:  # Unauthorized (normal failed login)
+                failed_attempts += 1
+                continue
+            else:
+                # Unexpected response
+                assert False, f"Unexpected response code: {response.status_code}"
```

Ruff (0.11.9)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)